service/src/test/java/bio/terra/cbas/initialization/TestCloneRecoveryServiceIntegration.java

```diff
@@ -22,6 +24,7 @@ public class TestCloneRecoveryServiceIntegration extends ContainerizedDatabaseTe
   @Autowired MethodDao methodDao;
   @Autowired RunSetDao runSetDao;
   @Autowired RunDao runDao;
+  @Autowired CloneRecoveryTransactionService transactionService;
```

Comment:
Now that this is `CloneRecoveryService` is doing all the calculations ahead of time, I think you should be able to unit-test that logic in unit tests, and mock-dao test the transaction level in a `CloneRecoveryTransactionServiceTest` to prove that it's calling down to the right low-level DB functions. Then the only "database integration" tests you need would be super thin "does the DB function do what it's supposed to when building SQL". And this whole class would probably be redundant